components/brave_ads/core/internal/serving/inline_content_ad_serving.cc

```diff
@@ -127,7 +126,6 @@ void InlineContentAdServing::ServeAd(
     return FailedToServeAd(ad.dimensions, std::move(callback));
   }
 
-  CHECK(eligible_ads_);
```

Comment:
`CHECK` is not required because we call `IsSupported` within the flow of `MaybeServeAd`